pilot/pkg/xds/eds.go

```diff
@@ -456,15 +456,18 @@ func shouldUseDeltaEds(req *model.PushRequest) bool {
 // onlyEndpointsChanged checks if a request contains *only* endpoints updates. This allows us to perform more efficient pushes
 // where we only update the endpoints that did change.
 func onlyEndpointsChanged(req *model.PushRequest) bool {
-	if len(req.ConfigsUpdated) > 0 {
-		for k := range req.ConfigsUpdated {
-			if k.Kind != kind.ServiceEntry {
-				return false
-			}
+	if len(req.ConfigsUpdated) == 0 {
+		return false
+	}
+	for cfg := range req.ConfigsUpdated {
+		if _, f := skippedEdsConfigs[cfg.Kind]; f {
+			continue
+		}
+		if cfg.Kind != kind.ServiceEntry {
+			return false
```

Comment:
I think we can also push inc when only DRs are changed? It seems not hard to implement as DR:CLA is basically 1:1, I will give it a try if you think it's worthy

Reply:
Same for delta CDS

Reply:
I would keep CDS and delta CDS separate, they are updated less frequently. EDS is most critical because pods start and go with much higher frequency than user-initiated config changes.

Reply:
Can't do it for anything that causes a CDS push. CDS push needs an EDS push after